citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -981,215 +1040,79 @@ mod tests {
-                    if let InternalServiceResponse::PeerRegisterSuccess { cid, peer_cid, username } = peer_response_packet {
-
-                        info!(target:"citadel", "User {cid} Registered Peer {peer_cid}");
-
-                        // Peer Connect
-                        let peer_connect_command = InternalServicePayload::PeerConnect {
-                            uuid: id,
-                            cid,
-                            username,
-                            peer_cid,
-                            peer_username: String::from("peer_test"),
-                            udp_mode: Default::default(),
-                            session_security_settings: Default::default(),
-                        };
-
-                        send(&mut sink, peer_connect_command).await?;
-                        let peer_connect_response = stream.next().await.unwrap()?;
-                        let peer_response_packet: InternalServiceResponse = bincode2::deserialize(&*next_packet)?;
-                        if let InternalServiceResponse::PeerConnectSuccess { cid} = peer_response_packet {
-
-                            info!(target:"citadel", "User {cid} Connected to Peer {peer_cid}");
-
-                            // Disconnect
-                            let disconnect_command = InternalServicePayload::Disconnect { uuid: id, cid };
-
-                            send(&mut sink, disconnect_command).await?;
-                            let next_packet = stream.next().await.unwrap()?;
-                            let response_disconnect_packet: InternalServiceResponse =
-                                bincode2::deserialize(&*next_packet)?;
-
-                            if let InternalServiceResponse::DisconnectSuccess { cid } =
-                                response_disconnect_packet
-                            {
-                                info!(target:"citadel", "Disconnected {cid}");
-                                Ok(())
-                            } else {
-                                panic!("Disconnection failed");
-                            }
-                        }
-                        else {
-                            panic!("Peer Connect Failed!");
-                        }
-                    }
-                    else {
-                        panic!("Peer Register Failed!");
-                    }
-                } else {
-                    panic!("Connection to server was not a success")
+        to_service_a
+            .send(InternalServicePayload::PeerRegister {
+                uuid: uuid_a,
+                cid: cid_a,
+                peer_id: cid_b.into(),
+            })
+            .unwrap();
+
+        to_service_b
+            .send(InternalServicePayload::PeerRegister {
+                uuid: uuid_b,
+                cid: cid_b,
+                peer_id: cid_a.into(),
+            })
+            .unwrap();
+
```

Comment:
on line 1079, we need to stop and wait for the internal service to tell us the register was a success :)